lib/phoenix_swagger/plug/validate_plug.ex

```diff
@@ -119,19 +135,70 @@ defmodule PhoenixSwagger.Plug.Validate do
     case Validator.validate(path, conn.body_params) do
       :ok -> :ok
       {:error, [{error, error_path} | _], _path} -> {:error, error, error_path}
-      {:error, error, error_path} ->  {:error, error, error_path}
+      {:error, error, error_path} -> {:error, error, error_path}
     end
   end
 
   defp equal_paths?([], []), do: true
-  defp equal_paths?([head | orig_path_rest], [head | req_path_rest]), do: equal_paths?(orig_path_rest, req_path_rest)
-  defp equal_paths?(["{" <> _ | orig_path_rest], [_ | req_path_rest]), do: equal_paths?(orig_path_rest, req_path_rest)
+
+  defp equal_paths?([head | orig_path_rest], [head | req_path_rest]),
+    do: equal_paths?(orig_path_rest, req_path_rest)
+
+  defp equal_paths?(["{" <> _ | orig_path_rest], [_ | req_path_rest]),
+    do: equal_paths?(orig_path_rest, req_path_rest)
+
   defp equal_paths?(_, _), do: false
 
   # It is pretty safe to strip request path by base path. They can't be
   # non-equal. In this way, the router even will not execute this plug.
   defp remove_base_path(path, []), do: path
+
   defp remove_base_path([_path | rest], [_base_path | base_path_rest]) do
     remove_base_path(rest, base_path_rest)
   end
-end
+
+  defp validate_header_params([]), do: :ok
```

Comment:
`validate_header_params/1` looks very similar to `validate_query_params/1`.
I think they can be combined into something like `validate_parameter_list`.

`validate_header_params/2` is fine, since it needs to pull the params from a different key in the `Conn` anyway.

Reply:
I'll try to fix this as soon as I can